src/fibsem_tools/io/h5.py

```diff
@@ -0,0 +1,57 @@
+import h5py
+from pathlib import Path
+from typing import Union, Any, Dict
+import warnings
+
+Pathlike = Union[str, Path]
+
+
+def partition_h5_kwargs(**kwargs) -> Dict[str, Any]:
+    """
+    partition kwargs into file-creation kwargs and dataset-creation kwargs
+    """
+    file_kwargs = kwargs.copy()
+    dataset_kwargs = {}
+    for key in ("shape", "dtype", "data", "chunks", "compressor"):
```

Comment:
```suggestion
    for key in ("name", "shape", "dtype", "data", "chunks", "compression", "compression_opts", "shuffle", "fletcher32", "external", "allow_unknown_filter"):
```

I just added some of the the relevant keywords here as well as correcting the the non-existent "compressor" keyword. Perhaps we should just add all of the known keywords?

Reply:
we could do this manually or automatically by inspecting the function signature the respective `h5py` functions, but i haven't played around with the latter approach so I don't know if it would be terrible